apps/web/src/components/Navigation/skip-to-code-editor.tsx

```diff
@@ -0,0 +1,16 @@
+'use client';
+import { usePathname } from 'next/navigation';
+
+export const SkipToCodeEditor = () => {
+  const pathname = usePathname();
+
+  if (!pathname.includes('challenge')) {
+    return null;
+  }
```

Comment:
Maybe checking the presence of a an element with id equal to `#code-editor` would be better?

Or go one step further and make this into a reusable `<SkipTo />` component.

Reply:
Used document.getElementById to check target element exists. Can't figure out how to defer the check until the page is fully loaded since Navigation loads first and the CodeEditor way later. I tried the following in useEffect:

- setTimeout with 0 delay
- adding event listener on window load
- setTimeout inside the event listener

I don't know how to enable a component to check if another component/element exists elegantly without something like a zustand store.

Reply:
use a useEffect / disable SSR for this specific component

Reply:
> use a useEffect / disable SSR for this specific component

```
  const pathname = usePathname();
  const [shouldShow, setShouldShow] = useState(false);

  useEffect(() => {
    if (pathname.includes('challenge')) {
      setShouldShow(true);
    }
  }, [pathname]);

  if (!shouldShow) {
    return null;
  }
```

Like this?

Reply:
oh nevermind misread your original comment